File: ProjectEuler/001-050/S29.cs
using System.Collections.Generic;
using System.Numerics;

namespace ProjectEuler
{
    public class S29 : ISolution
    {
        public string GetAnswer()
        {
            var results = new HashSet<BigInteger>();
            for (int i = 2; i <= 100; i++)
            {
                for (int j = 2; j <= 100; j++)
                {
                    results.Add(BigInteger.Pow(i, j));
                }
            }

            return results.Count.ToString();
        }
    }
}

File: src/001-050/S29.cc
#include "S29.h"
#include "../Utils.h"

using namespace std;

namespace ProjectEuler
{
std::string S29::GetAnswer()
{
	std::set<std::string> results;
	for (int i = 2; i <= 100; i++)
	{
		for (int j = 2; j <= 100; j++)
		{
			results.insert(Utils::Pow(i, j).toString());
		}
	}

	return std::to_string(results.size());
}

} // namespace ProjectEuler
